WordPress/Classes/ViewRelated/Post/PostMenuHelper.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import UIKit
+
+struct PostMenuHelper {
+
+    let statusViewModel: PostCardStatusViewModel
+
+    /// Creates a menu for post actions
+    ///
+    /// - parameters:
+    ///   - presentingView: The view presenting the menu
+    ///   - delegate: The delegate that performs post actions
+    func makeMenu(presentingView: UIView, delegate: InteractivePostViewDelegate) -> UIMenu {
+        let sections = makeSections(presentingView: presentingView, delegate: delegate)
+        return UIMenu(title: "", options: .displayInline, children: sections)
+    }
+
+    /// Creates post actions grouped into sections
+    ///
+    /// - parameters:
+    ///   - presentingView: The view presenting the menu
+    ///   - delegate: The delegate that performs post actions
+    private func makeSections(presentingView: UIView, delegate: InteractivePostViewDelegate) -> [UIMenu] {
+        return statusViewModel.buttonSections
+            .filter { !$0.buttons.isEmpty }
+            .map { section in
+                let actions = makeActions(for: section.buttons, presentingView: presentingView, delegate: delegate)
+                return UIMenu(title: "", options: .displayInline, children: actions)
+            }
+    }
+
+    /// Creates post actions
+    ///
+    /// - parameters:
+    ///   - buttons: The list of buttons to turn into post actions
+    ///   - presentingView: The view presenting the menu
+    ///   - delegate: The delegate that performs post actions
+    private func makeActions(
+        for buttons: [PostCardStatusViewModel.Button],
+        presentingView: UIView,
+        delegate: InteractivePostViewDelegate
+    ) -> [UIAction] {
+        let post = statusViewModel.post
+
+        return buttons.map { button in
+            UIAction(title: button.title(for: post), image: button.icon, attributes: button.attributes, handler: { _ in
+                button.performAction(for: post, view: presentingView, delegate: delegate)
```

Comment:
Not sure if there is a retain cycle, but I would suggest making the delegate `weak` just in case.

Reply:
Done! https://github.com/wordpress-mobile/WordPress-iOS/pull/21850/commits/67b80854d49baaeb0da32f3e811778e0f3f94fa1